src/alttxt/tokenmap.py

```diff
@@ -894,4 +891,93 @@ def calculate_intersection_trend(self) -> str:
             return f" The intersection sizes start from a value of {min_int_size} and then {intersection_trend} rise up to {max_int_size}."
         
     
+    def find_dominant_sets(self):
+        # Initialize a counter for all visible sets
+        set_occurrences = Counter()
+
+        # Go through each subset and count the occurrences of each visible set in subset names
+        for subset in self.data.subsets:
+            for set_name in self.grammar.visible_sets:
+                if set_name in subset.name:
+                    set_occurrences[set_name] += 1
+        # Find the most common sets, which returns a list of tuples (set_name, occurrences)
+        most_common_sets = set_occurrences.most_common(2)
+
+        # Extract only the names of the most and second most dominant sets
+        most_dominant_set = most_common_sets[0][0]
+        second_most_dominant_set = most_common_sets[1][0] if len(most_common_sets) > 1 else None
+
+        return f"{most_dominant_set}, and {second_most_dominant_set}"
+
+    def find_sets_in_large_subsets(self):
+        # Sort subsets by size in descending order
+        sorted_subsets = sorted(self.data.subsets, key=lambda subset: subset.size, reverse=True)
+        
+        # Extract set names from the 2nd largest subset
+        second_largest_sets = self.extract_set_names(sorted_subsets[1].name)
+        sets = []
+
+        if len(second_largest_sets) == 1:
+            third_largest_sets = self.extract_set_names(sorted_subsets[2].name)
+            # Find the intersection of sets between the 2nd and 3rd largest subsets
+            common_sets = second_largest_sets.union(third_largest_sets)
+
+            for cs in common_sets:
+                for set_name in self.grammar.visible_sets:
+                    print(set_name)
+                    if set_name in cs:
+                        sets.append(set_name)
+        
+        else:
+            for sm in second_largest_sets:
+                for set_name in self.grammar.visible_sets:
+                    if set_name in sm:
+                        sets.append(set_name)
+        
+         # Formatting the return value based on the size of the sets list
+        if len(sets) == 2:
+            return f"{sets[0]} and {sets[1]}"
+        elif len(sets) > 2:
+            return ', '.join(sets[:-1]) + ', and ' + sets[-1]
+        else:
+            return sets
+
+    def extract_set_names(self, subset_name):
+        set_names = set(subset_name.split(','))
+        return set_names
```

Comment:
Why not use the setMembership field? This feels like a workaround which shouldn't be needed.

Set Membership per subset should tell you exactly which sets compose the subset
ex: 
```json
"..." {
  "...",
  "setMembership": {
    "setA": "Yes",
    "setB": "No",
    "setC": "Yes",
  }
}

Reply:
added in this Commit: https://github.com/visdesignlab/upset-alt-txt-gen/pull/45/commits/ce0b2133200d73329f5e595c5a09f08b20f2f28a